File: 2021-08-15 (day28)/CXPhoenix/solution2.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        def binary(nl, target, left, right):
            if left > right:
                return -1
            mid = (left+right) // 2
            if target == nl[mid]:
                return mid
            if target > nl[mid]:
                return binary(nl, target, mid + 1, right)
            if target < nl[mid]:
                return binary(nl, target, left, mid - 1)
        i = 1
        count = 1
        while i < len(nums):
            index = binary(nums, nums[i], 0, i-1)
            if index > -1:
                del(nums[i])
                continue
            count += 1
            i += 1
        return count

File: 2021-08/2021-08-15 (day28)/Kevin/solution_2.js
var removeDuplicates = function (nums) {
    // Skip the first value in nums, so set `i` and `j` with `1` instead of `0`
    let i = 1;
    for (let j = 1; j < nums.length; j++) {
        if (nums[j] !== nums[j - 1]) {
            nums[i] = nums[j];
            i++;
        }
    }
    return i;
};
